PORTAL/jobs/queue.py

```diff
@@ -175,16 +193,10 @@ def _load(self):
         if fixed:
             with open(self._datafile, 'w') as outfile:
                 json.dump(data, outfile, indent=4)
-        data = self._data = JobQueueData(**data)
-        return data
-
-    def _save(self, data=None):
-        if data is None:
-            data = self._data
-        elif isinstance(data, types.SimpleNamespace):
-            if data is not self._data:
-                raise NotImplementedError
-            data = dict(vars(data))
+        self._data = JobQueueData(**data)
+        return self._data
+
+    def _save(self, data: JobQueueData) -> None:
```

Comment:
We still need a dict and to do the check:

```suggestion
    def _save(self, queuedata: JobQueueData) -> None:
        assert queuedata is self._data, (queuedata, self._data)
        data = dict(vars(queuedata))
```